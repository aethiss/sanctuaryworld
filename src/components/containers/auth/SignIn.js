import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

// Actions
import { setRegisterAccount } from '../../../redux/handlers/authActions'

// Forms
import {Formik, Form} from 'formik'
import * as Yup from 'yup'
import FormikFields from '../../commons/forms/FormikFields'

// Components
import BodyContainer from '../../commons/wrapper/BodyContainerResponsive'
import DiabloResponsiveButton from '../../commons/buttons/DiabloResponsiveButton/DiabloResponsiveButton'

const styles = {
  form: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  },
  button: {
    marginTop: '35px',
  },
}

@connect((state) => ({
  battletag: state.user?.battletag,
}), {
  actionRegister: setRegisterAccount,
})
class SignIn extends Component {

  render() {
    const { battletag, actionRegister } = this.props
    const backgroundColor = '#131210'

    if (!battletag) {
      return (
        <BodyContainer
          background='/images/background/act1-bg.jpg'
          contentOnCenter
        >
          <>
            <p>You need to be connected with your Blizzard Account</p>
            <DiabloResponsiveButton text='Connect' anchorLink='/api/connect/bnet' />
            <p>Secure Login with Blizzard Account</p>
          </>
        </BodyContainer>
      )
    }

    return (
      <BodyContainer
        background='/images/background/act1-bg.jpg'
        contentOnCenter
      >
        <Formik
          initialValues={{
            email: '',
            password: '',
            confirmPassword: '',
            acceptTerms: false,
          }}
          validationSchema={Yup.object({
            email: Yup.string()
              .email('Invalid email address')
              .required('Required'),
            password: Yup.string()
              .max(25, 'Must be 25 characters or less')
              .min(8, 'Must be 8 characters or more')
              .required('Required'),
            confirmPassword: Yup.string()
              .oneOf([Yup.ref('password'), null], 'Passwords must match')
              .required('Confirm Password is required'),
            acceptTerms: Yup.bool().oneOf([true], 'Accept Ts & Cs is required'),
          })}
          onSubmit={(values, {setSubmitting}) => {
            actionRegister({ ...values, username: battletag })
            setSubmitting(false)
          }}>
          {props => (
            <Form style={{...styles.form}}>
              <FormikFields
                name="email"
                type="text"
                label="Email"
                formikProps={props}
                icon="fa-at"
                backgroundColor={backgroundColor}
              />
              <FormikFields
                name="password"
                type="password"
                label="Password"
                formikProps={props}
                icon="fa-key"
                backgroundColor={backgroundColor}
              />
              <FormikFields
                name="confirmPassword"
                type="password"
                label="Confirm Password"
                formikProps={props}
                backgroundColor={backgroundColor}
              />
              <FormikFields
                name="acceptTerms"
                type="checkbox"
                label="Terms & Conditions"
                formikProps={props}
                backgroundColor={backgroundColor}
              />
              <div style={styles.button}>
                { !battletag ?
                  <DiabloResponsiveButton text="Account" minWidth={170} anchorLink='/account' /> :
                  <DiabloResponsiveButton text="Register" minWidth={170} />
                }
              </div>
            </Form>
          )}
        </Formik>
      </BodyContainer>
    )
  }

}

SignIn.propTypes = {
  battletag: PropTypes.string,
  actionRegister: PropTypes.func,
}

export default SignIn
