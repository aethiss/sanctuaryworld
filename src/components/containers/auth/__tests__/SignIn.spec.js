import React from 'react'
import { create, act } from 'react-test-renderer'
// import { act } from 'react-dom/test-utils'
import ReactDOM from 'react-dom'

import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
const mockStore = configureStore([])

import SignIn from '../SignIn'

const mockProps = {
  user: {
    battletag: 'pippo#1234',
    action: jest.fn(),
  }
}


describe('SignIn', () => {
  describe('Render', () => {
    it('[SignIn] : Render bnet logged user', () => {
      const store = mockStore(mockProps)
      const testRenderer = create(
        <Provider store={store}>
          <SignIn />
        </Provider>
      )
      expect(testRenderer).toMatchSnapshot()
    })
    it('[SignIn] : Render unauthenticated user', () => {
      const store = mockStore({
        battletag: false,
        action: jest.fn(),
      })
      const testRenderer = create(
        <Provider store={store}>
          <SignIn />
        </Provider>
      )
      expect(testRenderer).toMatchSnapshot()
    })
  })
  // TO-DO !!!
  describe('Events', () => {
    it.skip('[SignIn] : On click Register action is called', () => {
      const store = mockStore(mockProps)
      const testRenderer = create(
        <Provider store={store}>
          <SignIn />
        </Provider>
      )
      act(() => {
        console.log(testRenderer.root.findByType('button').props)
      })

      // expect(store.dispatch).toHaveBeenCalledTimes(1)
    })
  })
})
