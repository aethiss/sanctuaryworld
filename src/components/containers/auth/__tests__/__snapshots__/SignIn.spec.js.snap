// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn Render [SignIn] : Render bnet logged user 1`] = `
<div
  className="makeStyles-bodyWrapper-1"
  style={
    Object {
      "backgroundImage": "/images/background/act1-bg.jpg",
    }
  }
>
  <form
    action="#"
    onReset={[Function]}
    onSubmit={[Function]}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "marginTop": "30px",
          "width": "95%",
        }
      }
    >
      <label
        htmlFor="email"
        style={
          Object {
            "fontSize": "18px",
            "paddingBottom": "7px",
          }
        }
      >
        Email
      </label>
      <div
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <i
          className="fa fa-at"
          style={
            Object {
              "justifySelf": "center",
            }
          }
        />
        <input
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#131210",
              "border": "2px solid #ccc",
              "boxShadow": "0px 0px 9px 3px #ccc",
              "color": "#FFF",
              "height": "35px",
              "opacity": "0.7",
              "width": "275px",
            }
          }
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "marginTop": "30px",
          "width": "95%",
        }
      }
    >
      <label
        htmlFor="password"
        style={
          Object {
            "fontSize": "18px",
            "paddingBottom": "7px",
          }
        }
      >
        Password
      </label>
      <div
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <i
          className="fa fa-key"
          style={
            Object {
              "justifySelf": "center",
            }
          }
        />
        <input
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#131210",
              "border": "2px solid #ccc",
              "boxShadow": "0px 0px 9px 3px #ccc",
              "color": "#FFF",
              "height": "35px",
              "opacity": "0.7",
              "width": "275px",
            }
          }
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "marginTop": "30px",
          "width": "95%",
        }
      }
    >
      <label
        htmlFor="confirmPassword"
        style={
          Object {
            "fontSize": "18px",
            "paddingBottom": "7px",
          }
        }
      >
        Confirm Password
      </label>
      <div
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <input
          name="confirmPassword"
          onBlur={[Function]}
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#131210",
              "border": "2px solid #ccc",
              "boxShadow": "0px 0px 9px 3px #ccc",
              "color": "#FFF",
              "height": "35px",
              "opacity": "0.7",
              "width": "275px",
            }
          }
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "marginTop": "30px",
          "width": "95%",
        }
      }
    >
      <label
        htmlFor="acceptTerms"
        style={
          Object {
            "fontSize": "18px",
            "paddingBottom": "7px",
          }
        }
      >
        Terms & Conditions
      </label>
      <div
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <input
          checked={false}
          name="acceptTerms"
          onBlur={[Function]}
          onChange={[Function]}
          style={
            Object {
              "backgroundColor": "#131210",
              "opacity": "0.7",
            }
          }
          type="checkbox"
          value={false}
        />
      </div>
    </div>
    <div
      style={
        Object {
          "marginTop": "35px",
        }
      }
    >
      <button
        className="makeStyles-button-2"
      >
        <span
          className="makeStyles-buttonLeft-4"
        />
        <span
          className="makeStyles-buttonMid-3"
        >
          <span
            className="makeStyles-buttonText-6"
          >
            Register
          </span>
        </span>
        <span
          className="makeStyles-buttonRight-5"
        />
      </button>
    </div>
  </form>
</div>
`;

exports[`SignIn Render [SignIn] : Render unauthenticated user 1`] = `
<div
  className="makeStyles-bodyWrapper-1"
  style={
    Object {
      "backgroundImage": "/images/background/act1-bg.jpg",
    }
  }
>
  <p>
    You need to be connected with your Blizzard Account
  </p>
  <a
    href="/api/connect/bnet"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <button
      className="makeStyles-button-2"
    >
      <span
        className="makeStyles-buttonLeft-4"
      />
      <span
        className="makeStyles-buttonMid-3"
      >
        <span
          className="makeStyles-buttonText-6"
        >
          Connect
        </span>
      </span>
      <span
        className="makeStyles-buttonRight-5"
      />
    </button>
  </a>
  <p>
    Secure Login with Blizzard Account
  </p>
</div>
`;
